src/gleam/time/calendar.gleam

```diff
@@ -320,3 +322,14 @@ pub fn is_valid_time_of_day(time: TimeOfDay) -> Bool {
   && nanoseconds >= 0
   && nanoseconds <= 999_999_999
 }
+
+/// Compares two dates, returning an order.
+///
+pub fn naive_date_compare(one: Date, other: Date) -> Order {
+  use <- order.lazy_break_tie(int.compare(one.year, other.year))
+  use <- order.lazy_break_tie(int.compare(
+    month_to_int(one.month),
+    month_to_int(other.month),
+  ))
+  int.compare(one.day, other.day)
```

Comment:
Could you write this using pipes please? Slightly easier to read and results in less nesting